src/app/shared/components/number-input/number-input.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, Input, forwardRef } from '@angular/core';
+import {
+    NG_VALUE_ACCESSOR,
+    ControlValueAccessor,
+    NG_VALIDATORS,
+    FormControl,
+    Validator
+} from '@angular/forms';
+
+import { validateNumber } from './../../validators/number.validator';
+
+const CUSTOM_DATE_INPUT_VALUE_ACCESSOR = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => NumberInputComponent),
+    multi: true
+};
+
+const CUSTOM_DATE_INPUT_VALIDATOR = {
+    provide: NG_VALIDATORS,
+    useExisting: forwardRef(() => NumberInputComponent),
+    multi: true
+};
+
+@Component({
+   selector: 'number-input',
+   templateUrl: './number-input.component.html',
+   providers: [CUSTOM_DATE_INPUT_VALUE_ACCESSOR, CUSTOM_DATE_INPUT_VALIDATOR]
+})
+export class NumberInputComponent implements ControlValueAccessor, Validator {
+    @Input() nameOption: string;
+
+    private currentValue: any;
+
+    setValue(item) {
+        this.value = item.target.value;
+    }
+
+    set value(newValue) {
+        if (newValue !== this.currentValue) {
+            this.currentValue = newValue;
+            this.onChange(newValue);
+        }
+    }
+
+    get value() {
+        return this.currentValue;
+    }
+
+    onChange = (_) => {};
+    onTouched = () => {};
```

Comment:
No faktycznie, można by to tak zrobić, umówmy się, że w prawdziwym projekcie będę o tym pamiętał ;)